### PR TITLE
Backend Saved Connections

### PR DESCRIPTION
**Pull request by larahynkim**

### CODE REVIEW

File: back-end/users.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  }
+})
+
+// Route for updating user information
+router.put('/:id', (req, res) => {
+  try {
+    // Update user information
+    // Simulate updating the database with only these fields
+    const userId = req.params.id
+    const { email, first_name: firstName, last_name: lastName, url_picture: urlPicture } = req.body
```
Comment (SivanC) [inline back-end/users.js:59]: no key for email?

File: front-end/src/ConnectionDetails.js [hunk 2]
```diff
@@ -1,61 +1,59 @@
...
   useEffect(() => {
-    async function fetchImage() {
+    const fetchUserDetails = async () => {
       try {
-        const logoUrl = 'https://picsum.photos/200/300';
-        const response = await axios.get(logoUrl);
-        if (response.status === 200) {
-          setImageUrl(logoUrl);
-        }
+        const response = await axios.get(`http://localhost:3001/api/connection-details/${id}`);
```
Comment (SivanC) [inline front-end/src/ConnectionDetails.js:14]: I think we are supposed to access this URL through the .env file?

File: front-end/src/SavedConnections.js [hunk 3]
```diff
@@ -9,13 +9,13 @@ const SavedConnections = () => {
   useEffect(() => {
     const fetchData = async () => {
       try {
-        const response = await axios.get('https://api.mockaroo.com/api/1a199910?count=16&key=14ea8470');
+        const response = await axios.get('http://localhost:3001/api/saved-connections');
```
Comment (SivanC) [inline front-end/src/SavedConnections.js:12]: Should also be fetched from .env

File: back-end/connections.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+const express = require('express');
+const axios = require('axios');
+const router = express.Router();
+
+router.get('/saved-connections', async (req, res) => {
+  try {
+    const mockApiUrl = 'https://api.mockaroo.com/api/1a199910?count=16&key=14ea8470';
```
Comment (SivanC) [inline back-end/connections.js:7]: get from .env file

File: back-end/test/connections.test.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+const chai = require('chai');
+const chaiHttp = require('chai-http');
+const server = require('../app'); 
+const should = chai.should();
+
+chai.use(chaiHttp);
+
+describe('Connections', () => {
+  describe('/GET saved-connections', () => {
+    it('it should GET all the connections', (done) => {
```
Comment (SivanC) [inline back-end/test/connections.test.js:10]: I think you need to test:
- When the request is successful, does it return the expected data
- When the request is not successful, does it behave appropriately
- Does the request fail/succeed as expected, i.e. is it failing where it should fail and succeeding where it should succeed?

Ultimately you can't guarantee that it won't fail, you just want to make sure that no matter what it is doing what you expect.

Also I think you don't need to write 'it' again, but ultimately inconsequential.

**Review by SivanC — CHANGES_REQUESTED** [review]
Lara since this is your PR I mostly just addressed the changes you made and not so much Byron's. Looking good overall, just need to add a couple tests and also get your API information from the .env file instead of inside the code. Let me know if you have questions.
--Sivan

File: back-end/test/connections.test.js
Comment (SivanC) [inline back-end/test/connections.test.js:1]: Also maybe rename to connections-test.js?

File: front-end/src/ConnectionDetails.js (hunk 2 above)
Comment (larahynkim) [reply front-end/src/ConnectionDetails.js:14]: I wasn't responsible for the ConnectionDetails page, should I still try to fix it or should I leave it?

File: front-end/src/ConnectionDetails.js [hunk 6]
```diff
@@ -1,59 +1,61 @@
...
+const ConnectionDetails = ({ ScannedInfo }) => {
+  const [imageUrl, setImageUrl] = useState('');
 
   useEffect(() => {
-    const fetchUserDetails = async () => {
+    async function fetchImage() {
       try {
-        const response = await axios.get(`http://localhost:3001/api/connection-details/${id}`);
-        setUserDetails(response.data);
+        const logoUrl = 'https://picsum.photos/200/300';
```
Comment (SivanC) [inline front-end/src/ConnectionDetails.js:13]: I think we'll need to grab profile pictures through a route no?

**Comment by Pan-LLi** [comment]
Hi Lara, I run your code on CodeSpace, it seems that fetching data in your SavedConnections.js is not working. Can you check that?

**Comment by Pan-LLi** [comment]
You can edit your CSS style of displaying different connections, when the connection goes out of screen/browser size when setting the page to size of the phone.

**Pull request merged**